### PR TITLE
add resample_pseudodata flag

### PR DESCRIPTION
**Pull request by RoyStegeman**
For parameter fits it is important not to distort the pseudodata sampling distribution. This PR adds a flag `resample_negative_pseudodata` that can be set to False to also accept negative pseudodata values instead of resampling.

### CODE REVIEW

File: validphys2/src/validphys/pseudodata.py [hunk 1]
```diff
@@ -263,7 +268,10 @@ def make_replica(
         # Shifting pseudodata
         shifted_pseudodata = (all_pseudodata + shifts) * mult_part
         # positivity control
-        if np.all(shifted_pseudodata[full_mask] >= 0):
+        if resample_pseudodata:
+            if np.all(shifted_pseudodata[full_mask] >= 0):
+                return shifted_pseudodata
+        else:
```
Comment (scarlehoff) [inline validphys2/src/validphys/pseudodata.py:274]:
```suggestion
        if np.all(shifted_pseudodata[full_mask] >= 0) or not resample_negative_pseudodata:
```

File: validphys2/src/validphys/pseudodata.py [hunk 2]
```diff
@@ -129,6 +129,7 @@ def make_replica(
     sep_mult,
     genrep=True,
     max_tries=int(1e6),
+    resample_pseudodata=True,
```
Comment (scarlehoff) [inline validphys2/src/validphys/pseudodata.py:132]:
```suggestion
    resample_negative_pseudodata=True,
```

I think this way it is clearer what is happening.

File: validphys2/src/validphys/pseudodata.py [hunk 3]
```diff
@@ -163,6 +164,10 @@ def make_replica(
         If after max_tries (default=1e6) no physical configuration is found,
         it will raise a :py:class:`ReplicaGenerationError`
 
+
+    resample_pseudodata: bool
+        Set to False to allow for negative predictions and avoid resampling
+
```
Comment (scarlehoff) [inline validphys2/src/validphys/pseudodata.py:170]:
```suggestion
    resample_negative_pseudodata: bool
        When True, replicas that produce negative predictions will be resampled for ``max_tries`` until all points are positive (default: True)
```

**Review by scarlehoff — COMMENTED** [review]
I guess this will be a runcard parameter?

Could you add it either to docs or to some runcard example (even with the default value) just to signal that it exists.

**Comment by RoyStegeman** [comment]
This doesn't do any resampling and just accepts also negative xsecs

**Review by scarlehoff — APPROVED** [review]

**Comment by andreab1997** [comment]
> This doesn't do any resampling and just accepts also negative xsecs

Ok yes but I understood you did try the resampling no?

**Comment by RoyStegeman** [comment]
Yes I did do MHOU fits with resampling (so the same as in master now), but I didn't track how much time that took

**Pull request merged**